x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/types.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+interface BaseData {
+  analyticsUnavailable: boolean;
+  allTags: string[];
+  // NOTE: The API sends us back even more data than this (e.g.,
+  // startDate, endDate, currentTag, logRetentionSettings, query),
+  // but we currently don't need that data in our front-end code,
+  // so I'm opting not to list them in our types
+}
+
+export interface AnalyticsData extends BaseData {
+  recentQueries: RecentQuery[];
```

Comment:
I don't know if this actually matters because of hoisting. It's probably just a subjective / organizational / code style type comment.

I try to avoid using types in other types until after they are defined. So I would have put `RecentQuery` and `Query` above this.

Food for thought.

Reply:
No strong pref either way, will reorder 👍

Reply:
4400676